internal/deployments.go

```diff
@@ -554,6 +567,7 @@ func (i *Internal) getDeployment(job *model.Job) (*appsv1.Deployment, error) {
 					Volumes:                      i.deploymentVolumes(job),
 					InitContainers:               i.initContainers(job),
 					Containers:                   i.deploymentContainers(job),
+					ImagePullSecrets:             i.imagePullSecrets(job),
```

Comment:
https://pkg.go.dev/k8s.io/api/core/v1#PodSpec.ImagePullSecrets should be what I'm setting here, I think